### PR TITLE
add User Unsubscription and Querying endpoints

### PR DESCRIPTION
**Pull request by SergiLomtadze**

### CODE REVIEW

File: src/ExadelMentorship.BusinessLogic/Features/Reports/UserQuery/UserQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+namespace ExadelMentorship.BusinessLogic.Features.Reports.UserQuery
+{
+    public class UserQueryHandler : ICommandHandler<UserQuery, IEnumerable<ReportUser>>
+    {
+        private IReportUserRepo _reportUserRepo;
+        public UserQueryHandler(IReportUserRepo reportUserRepo)
+        {
+            _reportUserRepo = reportUserRepo;
+        }
+        public async Task<IEnumerable<ReportUser>> Handle(UserQuery command)
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.BusinessLogic/Features/Reports/UserQuery/UserQueryHandler.cs:18]: Don't do unnecessary coding.

`return _reportUserRepo.GetAll().AsEnumerable()`

File: src/ExadelMentorship.BusinessLogic/Features/Reports/UserSubscription/UserSubscriptionCommand.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 namespace ExadelMentorship.BusinessLogic.Features.Reports.UserSubscription
 {
-    public class UserSubscriptionCommand : ICommand<string>
+    public class UserUnSubscriptionCommand : ICommand<string>
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.BusinessLogic/Features/Reports/UserSubscription/UserSubscriptionCommand.cs:8]: `UnsubscribeUserCommand`

File: src/ExadelMentorship.Persistence/ReportUserRepo.cs [hunk 3]
```diff
@@ -31,5 +32,10 @@ public async Task SaveAsync(string userName, string email)
             });
             await _dbContext.SaveChangesAsync();
         }
+
+        public void Delete(int userId)
+        {
+            _dbContext.Remove(userId);
```
Comment (kdavletov-exadel) [inline src/ExadelMentorship.Persistence/ReportUserRepo.cs:38]: I'm not sure if it is correct.

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.Persistence/ReportUserRepo.cs (hunk 3 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.Persistence/ReportUserRepo.cs:38]: done

File: src/ExadelMentorship.BusinessLogic/Features/Reports/UserSubscription/UserSubscriptionCommand.cs (hunk 2 above)
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Features/Reports/UserSubscription/UserSubscriptionCommand.cs:8]: done

**Review by kdavletov-exadel — CHANGES_REQUESTED** [review]

File: src/ExadelMentorship.BusinessLogic/Features/Reports/UserQuery/UserQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        {
+            return await Task.FromResult(_reportUserRepo.GelAll().AsEnumerable());
```
[the remark this answers is not in the document]
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Features/Reports/UserQuery/UserQueryHandler.cs:20]: return (Task<IEnumerable<ReportUser>>)_reportUserRepo.GelAll().AsEnumerable(); ??

[the remark this answers is not in the document]
Comment (kdavletov-exadel) [reply src/ExadelMentorship.BusinessLogic/Features/Reports/UserQuery/UserQueryHandler.cs:20]: Some options:
1. No async/await keywords.
2. Change return type to `Task<ReportUser[]>` and return `.GetAll().ToArrayAsync()`.
3. Utilize IAsyncEnumerable

[the remark this answers is not in the document]
Comment (SergiLomtadze) [reply src/ExadelMentorship.BusinessLogic/Features/Reports/UserQuery/UserQueryHandler.cs]: done

**Pull request merged**